### PR TITLE
Fix multi dimensional arrays with FlattenedStorage.to_pandas

### PR DESCRIPTION
**Pull request by pmrv**

### CODE REVIEW

**Review by jan-janssen — CHANGES_REQUESTED** [review]
To guarantee the stability of these kinds of conversions it would be great to add tests.

**Comment by jan-janssen** [comment]
@pmrv Can you merge `main` into your branch? Especially as the directory structure for the tests changed?

**Comment by pmrv** [comment]
> @pmrv Can you merge `main` into your branch? Especially as the directory structure for the tests changed?

Should be fixed now.

File: pyiron_base/_tests.py [hunk 1]
```diff
@@ -43,9 +43,9 @@ class PyironTestCase(unittest.TestCase, ABC):
     def setUp(self):
         self.addTypeEqualityFunc(np.ndarray, self._assert_equal_numpy)
 
-    def _assert_equal_numpy(self, a, b, msg=None):
+    def _assert_equal_numpy(self, a, b, msg=""):
```
Comment (jan-janssen) [inline pyiron_base/_tests.py]: Is this necessary? I thought with the changes in line 48 now `None` is supported and maybe the better choice.

File: tests/unit/storage/test_flattenedstorage.py [hunk 2]
```diff
@@ -945,8 +963,21 @@ def test_to_pandas(self):
...
                     )
 
-        dfe = store.to_pandas(explode=True)
+        try:
+            dfe = store.to_pandas(explode=True)
+        except:
+            # Regression test
+            self.fail(
+                "to_pandas must not fail when multidimensional arrays are present!"
+            )
```
Comment (jan-janssen) [inline tests/unit/storage/test_flattenedstorage.py]: I am a bit confused about the `try except` part. I agree that it gives a nicer error message but at the same time it is not as readable.

File: pyiron_base/_tests.py (hunk 1 above)
Comment (pmrv) [inline pyiron_base/_tests.py]:
```suggestion
    def _assert_equal_numpy(self, a, b, msg=None):
```

**Review by jan-janssen — APPROVED** [review]
Looks good to me

**Pull request merged**